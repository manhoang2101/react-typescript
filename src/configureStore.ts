import { Store, createStore, applyMiddleware } from "redux";
import createSagaMiddleware from "redux-saga";
import { routerMiddleware } from "connected-react-router";
import { composeWithDevTools } from "redux-devtools-extension";
import { History } from "history";
import { createRootReducer, rootSaga } from "./stories";
import { IAppState } from "./stories/types";

export default function configureStore(history: History): Store<IAppState> {
  const composeEnhancers = composeWithDevTools({});
  const sagaMiddleware = createSagaMiddleware();
  const stories = createStore(
    createRootReducer(history),
    composeEnhancers(applyMiddleware(routerMiddleware(history), sagaMiddleware))
  );
  sagaMiddleware.run(rootSaga);
  return stories;
}
