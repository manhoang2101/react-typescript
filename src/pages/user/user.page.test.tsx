import configureStore from "redux-mock-store";
import React from "react";
import { UserPage, mapDispatchToProps } from ".";
import { Provider } from "react-redux";
import { shallow } from "enzyme";
import { render } from "@testing-library/react";
import EUserAction from "../../stories/user/user.actions";

const initialState = {
  users: [],
  user: null,
};
const mockStore = configureStore();
let stories: any;
describe("UserPage", () => {
  beforeEach(() => {
    stories = mockStore(initialState);
  });

  test("UserPage as component", () => {
    const container = shallow(
      <Provider store={stories}>
        <UserPage />
      </Provider>
    );
    expect(container.find(UserPage).length).toBe(1);
  });

  it("should call addUser", () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).addUser({ id: 100, name: "name" });
    expect(dispatch.mock.calls[0][0]).toEqual({
      payload: { id: 100, name: "name" },
      type: EUserAction.ADD_USER,
    });
  });
  it("should call setUser", () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).setUser({ id: 100, name: "name" });
    expect(dispatch.mock.calls[0][0]).toEqual({
      payload: { id: 100, name: "name" },
      type: EUserAction.SET_USER,
    });
  });
});
