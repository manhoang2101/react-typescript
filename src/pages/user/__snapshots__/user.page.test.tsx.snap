// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserPage render component 1`] = `
<div>
  <div
    class="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <table
      aria-label="simple"
      class="MuiTable-root AppTable-table-2 UserContainer-table-1"
    >
      <thead
        class="MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root tr-thead MuiTableRow-head"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head id AppTable-tdThead-6 td-thead"
            scope="col"
          >
            id
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head name AppTable-tdThead-6 td-thead"
            scope="col"
          >
            Name
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head cardNumber AppTable-tdThead-6 td-thead"
            scope="col"
          >
            cardNumber
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head cardType AppTable-tdThead-6 td-thead"
            scope="col"
          >
            cardType
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root"
      />
    </table>
  </div>
</div>
`;
